spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/StreamDefinitionController.java

```diff
@@ -190,6 +190,53 @@ public void delete(@PathVariable("name") String name) {
 	}
 
 	/**
+	 * Return a list of related stream definition resources based on the given stream name.
+	 * Related streams include the main stream and the tap stream(s) on the main stream.
+	 *
+	 * @param name the name of an existing stream definition (required)
+	 */
+	@RequestMapping(value = "/{name}/related", method = RequestMethod.GET)
+	@ResponseStatus(HttpStatus.OK)
+	public PagedResources<StreamDefinitionResource> listRelated(@PathVariable("name") String name,
+																@RequestParam("nested") boolean nested,
+																PagedResourcesAssembler<StreamDefinition> assembler) {
+		Set<StreamDefinition> relatedDefinitions = new HashSet<>();
+		StreamDefinition currentStreamDefinition = repository.findOne(name);
+		if (currentStreamDefinition == null) {
+			throw new NoSuchStreamDefinitionException(name);
+		}
+		Iterable<StreamDefinition> definitions = repository.findAll();
+		List<StreamDefinition> result = new ArrayList<>(findRelatedDefinitions(currentStreamDefinition, definitions,
+				relatedDefinitions, nested));
+		return assembler.toResource(new PageImpl<>(result), streamDefinitionAssembler);
+	}
+
+	private Set<StreamDefinition> findRelatedDefinitions(StreamDefinition currentStreamDefinition, Iterable<StreamDefinition> definitions,
+														  Set<StreamDefinition> relatedDefinitions, boolean nested) {
+		relatedDefinitions.add(currentStreamDefinition);
+		for (StreamDefinition definition: definitions) {
+			String tapNamePrefix = ":" + currentStreamDefinition.getName();
```

Comment:
> This is very String oriented. Can we work with the object model instead (parse result or something)?

May be. But, it turned out that using the simple `String` based check against the DSL text is easier for the tap prefix check(Given we have more deterministic way of specifying this) than the StreamParser approach. 
Do you see any issue?